citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -979,6 +972,27 @@ protected void doPostJsonResponse(ParameterParser params,
 		return results;
 	}
 
+	protected Map<String, Object> addSection(ParameterParser params, SessionState state) {
+		String message;
+		Map<String, Object> results = new HashMap<String, Object>();
+		try {
+			String addSectionHTML = params.getString("addSectionHTML");
```

Comment:
This needs doing before we release but is it okay to put it in a different pull request as I just want to get the basic editing functionality working first and then fix security issues like this?